phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewConcurrencyAndFailureIT.java

```diff
@@ -0,0 +1,793 @@
+            getUtility().getHBaseCluster().getMaster().balanceSwitch(false);
+            splitSystemCatalog();
+        }
+    }
+
+    @After
+    public void cleanup() {
+        latch1 = null;
+        latch2 = null;
+        throwExceptionInChildLinkPreHook = false;
+        slowDownAddingChildLink = false;
+    }
+
+    @Test
+    public void testChildViewCreationFails() throws Exception {
+        try (Connection conn = DriverManager.getConnection(getUrl());
+                Statement stmt = conn.createStatement()) {
+
+            String fullTableName = SchemaUtil.getTableName(SCHEMA1,
+                    generateUniqueName());
+            String failingViewName = FAILED_VIEWNAME;
+            String succeedingViewName = SchemaUtil.getTableName(SCHEMA3,
+                    generateUniqueName());
+
+            String createTableDdl = "CREATE TABLE " + fullTableName
+                    + "  (k INTEGER NOT NULL PRIMARY KEY, v1 DATE)"
+                    + tableDDLOptions;
+            stmt.execute(createTableDdl);
+
+            String createViewDdl = "CREATE VIEW " + failingViewName
+                    + " (v2 VARCHAR) AS SELECT * FROM "
+                    + fullTableName + " WHERE k > 5";
+            try {
+                stmt.execute(createViewDdl);
+                fail();
+            } catch (PhoenixIOException ignored) {
+            }
+            createViewDdl = "CREATE VIEW " + succeedingViewName
+                    + "(v2 VARCHAR) AS SELECT * FROM " + fullTableName
+                    + " WHERE k > 10";
+            stmt.execute(createViewDdl);
+
+            // the first child view should not exist
+            try {
+                PhoenixRuntime.getTableNoCache(conn, failingViewName);
+                fail();
+            } catch (TableNotFoundException ignored) {
+            }
+
+            // we should be able to load the table
+            PhoenixRuntime.getTableNoCache(conn, fullTableName);
+            // we should be able to load the second view
+            PhoenixRuntime.getTableNoCache(conn, succeedingViewName);
+        }
+    }
+
+    @Test
+    public void testConcurrentViewCreationAndTableDrop() throws Exception {
+        try (Connection conn = DriverManager.getConnection(getUrl());
+                Statement stmt = conn.createStatement()) {
+            String fullTableName = SchemaUtil.getTableName(SCHEMA1,
+                    generateUniqueName());
+            String fullViewName1 = SLOW_VIEWNAME_PREFIX + "_"
+                    + generateUniqueName();
+            latch1 = new CountDownLatch(1);
+            latch2 = new CountDownLatch(1);
+            String tableDdl = "CREATE TABLE " + fullTableName +
+                    "  (k INTEGER NOT NULL PRIMARY KEY, v1 INTEGER, v2 INTEGER)"
+                    + tableDDLOptions;
+            stmt.execute(tableDdl);
+
+            ExecutorService executorService = Executors.newFixedThreadPool(
+                    1, new ThreadFactory() {
```

Comment:
nit: maybe use `newSingleThreadExecutor()` instead? (for all similar places in this file)